pkg/hierarchy/manager.go

```diff
@@ -38,11 +38,31 @@ func NewManager[CQ clusterQueueNode[C], C cohortNode[CQ, C]](newCohort func(stri
 }
 
 func (m *Manager[CQ, C]) AddClusterQueue(cq CQ) {
-	m.ClusterQueues[cq.GetName()] = cq
+	m.clusterQueues[cq.GetName()] = cq
+}
+
+func (m *Manager[CQ, C]) GetClusterQueue(name string) CQ {
```

Comment:
```suggestion
func (m *Manager[CQ, C]) ClusterQueue(name string) CQ {
```
nit: There is no strict rule on Getters in golang, but generally the "Get" prefix can be skipped: https://go.dev/doc/effective_go#Getters